tests/test_weighted_mean_estimator.py

```diff
@@ -1,787 +1,187 @@
-                    1.05165081e-02,
-                    3.30510143e-03,
-                    -3.48652120e-03,
-                    -4.23228797e-04,
-                    1.40484061e-02,
-                    1.42914291e-03,
-                    -1.28129504e-02,
-                    2.19868825e-03,
-                    -6.30835037e-03,
-                    1.18524223e-03,
-                    -2.97855052e-02,
-                    1.15491057e-03,
-                    -8.27947006e-03,
-                    3.45442781e-03,
-                    -4.72868856e-03,
-                    2.66615329e-03,
-                    -7.87929790e-03,
-                    8.84126590e-04,
-                    1.59402808e-03,
-                    -9.06854048e-05,
-                    -8.79119004e-03,
-                    1.76449039e-03,
-                    -1.36414673e-02,
-                    1.56793855e-03,
-                    1.44708445e-02,
-                    -2.55974802e-03,
-                    5.38506357e-03,
-                    -3.24188673e-03,
-                    4.81582945e-04,
-                    7.74260101e-05,
-                    5.48772082e-03,
-                    1.92058500e-03,
-                    -4.63538896e-03,
-                    -2.02735133e-03,
-                    3.67592386e-03,
-                    7.23486969e-04,
-                    1.81838422e-03,
-                    1.78793284e-03,
-                    -8.01474060e-03,
-                    -8.54007528e-03,
-                    1.96353845e-03,
-                    -2.16254252e-03,
-                    -3.64243996e-05,
-                    -2.27329863e-03,
-                    1.11424393e-03,
-                    -1.39389189e-03,
-                    2.57787159e-04,
-                    3.66918811e-03,
-                    1.31477774e-03,
-                    6.82220128e-04,
-                    1.41822851e-03,
-                    -1.89476924e-03,
-                    -6.43966255e-05,
-                    -7.87888465e-04,
-                    -6.99459279e-04,
-                    1.08918981e-03,
-                    2.25264584e-03,
-                    -1.43651015e-04,
-                    7.68377620e-04,
-                    5.05955256e-04,
-                    2.66936132e-06,
-                    2.24934884e-03,
-                    6.70529439e-04,
-                    4.81121742e-04,
-                    -6.40789745e-05,
-                    -3.35915672e-04,
-                    -7.98651783e-04,
-                    -9.82705453e-04,
-                    6.46337066e-05,
-                ]
-            ]
-            * self.r
+@pytest.fixture(scope="module")
+def weights(sim):
+    # Construct simple test weights;
+    # one uniform positive and negative weighted volume respectively.
+    r = 2  # Number of weighted volumes
+    weights = np.ones((sim.n, r)) / np.sqrt(sim.n)
+    weights[:, 1] *= -1  # negate second weight vector
+
+    return weights
+
+
+@pytest.fixture(
```

Comment:
Shouldn't this define a `preconditioner` fixture?

Reply:
None is the default "circulant".  `"None"` is no preconditioner .  This is confusing and the codes should probably be rewritten, but I was only trying to fix the reported bug.

Reply:
Great. Thanks!